### PR TITLE
chore: security updates

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

Provide security patches mainly around CI 

## Related Issue

Fixes #

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/pepr/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: .github/workflows/commitlint.yml [hunk 1]
```diff
@@ -24,4 +25,4 @@ jobs:
         run: npm install --save-dev @commitlint/{config-conventional,cli}
 
       - name: Lint PR title
-        run: echo "${{ github.event.pull_request.title }}" | npx commitlint
+        run: echo "${{ github.event.pull_request.title }}" | npx commitlint@v17.7.2
```
Comment (btlghrants) [inline .github/workflows/commitlint.yml]: Maybe I'm just unfamiliar with the syntax here, but it looks like we're `npm install`ing commit lint at no-particular-version (so, latest..?) but then we're specifying v17.7.2 when we run it.  Is that right?

What happens when a new version of commitlint gets released?  Wouldn't this `npm install` the newer version but then _also_ try to run commit lint with @v17.7.2?

Comment (cmwylie19) [reply .github/workflows/commitlint.yml]: I believe that is correct, good catch. Maybe we can take out that `npm i` completely? I don't think it is doing anything

Comment (btlghrants) [reply .github/workflows/commitlint.yml]: Or, wait, I guess `npm install` is just going to install the version we have stored in the package-lock.json file, innit?  In that case, _why_ do we need to specify the version in our workflow file too?

Comment (cmwylie19) [reply .github/workflows/commitlint.yml]: this is resolved together through a conversation in gather.

Comment (btlghrants) [inline .github/workflows/commitlint.yml]:
```suggestion
        run: echo "${{ github.event.pull_request.title }}" | npx commitlint
```
We decided that we shouldn't need this cuz it's spec'd in the package-lock.json file, right?

**Review by btlghrants — APPROVED** [review]
Looks good; just like we talked 'bout.

**Pull request merged**